test/end-to-end/offer.js

```diff
@@ -633,41 +632,139 @@ describe("Offer endpoint tests", () => {
+                    company: test_company._id
+                });
+                [test_1, test_2, test_3] =
+                    await Promise.all([test_offer_1, test_offer_2, test_offer_3].map(async (offer) => {
+                        offer.owner = test_company._id.toString();
+                        const { _id } = await Offer.create(offer);
+                        offer["_id"] = _id.toString();
+                        return offer;
+                    }));
+                // test_user_non_admin = {
+                //     email: "company@email.com",
+                //     password: "password123",
+                // };
+            });
+
+            test("should return offer", async () => {
+                let res = await test_agent.get(`/offers/${test_1["_id"]}`);
+                expect(res.status).toBe(HTTPStatus.OK);
+                const extracted_data_1 = res.body;
+
+                res = await test_agent.get(`/offers/${test_2["_id"]}`);
+                expect(res.status).toBe(HTTPStatus.OK);
+                const extracted_data_2 = res.body;
+
+                expect(extracted_data_1).toMatchObject(test_1);
+                expect(extracted_data_2).toMatchObject(test_2);
+            });
+
+            test("should fail if not admin or owner", async () => {
+                const res = await test_agent.get(`/offers/${test_3["_id"]}`);
+                expect(res.status).toBe(HTTPStatus.NOT_FOUND);
             });
 
-            test("Offer 1", async () => {
-                const res = await request().get(`/offers/${test_offer_1.id}`);
+            test("should return hidden offer as admin", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_admin)
+                    .expect(200);
+
+                const res = await test_agent.get(`/offers/${test_3["_id"]}`);
                 expect(res.status).toBe(HTTPStatus.OK);
-                expect(res.body).toBe(test_offer_1);
+
+                const extracted_data = res.body;
+
+                expect(extracted_data).toMatchObject(test_3);
             });
 
-            test("Offer 1", async () => {
-                const res = await request().get(`/offers/${test_offer_1.id}`);
+            test("should return hidden offer as company", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_company)
+                    .expect(200);
+
+                const res = await test_agent.get(`/offers/${test_3["_id"]}`);
                 expect(res.status).toBe(HTTPStatus.OK);
-                expect(res.body).toBe(test_offer_1);
+                const extracted_data = res.body;
+
+                expect(extracted_data).toMatchObject(test_3);
+            });
+
+            afterAll(async () => {
+                await Company.deleteMany({});
+                await Offer.deleteMany({});
             });
```

Comment:
Unnecessary, each describe does its setup in the beforeAll (cleaning existing stuff and creating whatever)

But if it was necessary, it should be before any test:

describe
\ - beforeAll
\ - afterAll
\ - beforeEach
\ - afterEach
\ - tests